src/main/java/com/hades/social/SocialController.java

```diff
@@ -0,0 +1,57 @@
+package com.hades.social;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.social.connect.Connection;
+import org.springframework.social.connect.UserProfile;
+import org.springframework.social.facebook.api.Facebook;
+import org.springframework.social.facebook.connect.FacebookConnectionFactory;
+import org.springframework.social.oauth2.AccessGrant;
+import org.springframework.social.oauth2.GrantType;
+import org.springframework.social.oauth2.OAuth2Operations;
+import org.springframework.social.oauth2.OAuth2Parameters;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.hades.annotation.Get;
+import com.hades.annotation.PermitEndpoint;
+
+@Component
+@RestController
+public class SocialController {
+	
+	@Autowired
+	private SocialFacade socialFacade;
+	
+	@Value("${facebook.clientId}")
+	private String facebookClientId;
+	
+	@Value("${facebook.clientSecret}")
+	private String facebookClientSecret;
+	
+	@Get("/facebook")
+	@PermitEndpoint
+	@ResponseStatus(HttpStatus.ACCEPTED)
+	public String facebookLinkGenerator(){
+		FacebookConnectionFactory connectionFactory = new FacebookConnectionFactory(facebookClientId, facebookClientSecret);
+		OAuth2Operations oauthOperations = connectionFactory.getOAuthOperations();
+		OAuth2Parameters params = new OAuth2Parameters();
+		params.setScope("email");
+		params.setRedirectUri("https://dev.qualfacul.com:6660/facebook/login");
```

Comment:
Eu deixei dev porque não temos o site no ar ainda. Quando o site for para o ar o link será alterado para www. Só compensaria criar uma property para ele se fosse para informar o domínio completo (www.qualfacul.com), não?